crates/forge/tests/integration/interact_with_state.rs

```diff
@@ -0,0 +1,84 @@
+use forge_runner::forge_config::ForgeTrackedResource;
+use indoc::indoc;
+use test_utils::runner::{Contract, assert_passed};
+use test_utils::running_tests::run_test_case;
+use test_utils::test_case;
+
+#[test]
+#[cfg_attr(not(feature = "interact-with-state"), ignore)]
+fn get_contract_address_in_interact_with_state() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use snforge_std::{
+            ContractClassTrait, DeclareResultTrait, declare, interact_with_state, test_address,
+        };
+        use starknet::{ContractAddress, get_contract_address};
+
+        #[starknet::interface]
+        trait IEmpty<TContractState> {
+            fn get_address(ref self: TContractState) -> ContractAddress;
+        }
+
+        #[test]
+        fn test_contract_address_set_correctly() {
+            let contract = declare("Empty").unwrap().contract_class();
+            let (contract_address, _) = contract.deploy(@array![]).unwrap();
+            let (other_empty_contract, _) = contract.deploy(@array![]).unwrap();
+            let dispatcher = IEmptyDispatcher { contract_address };
+
+            let assert_eq_addresses = |a: ContractAddress, b: ContractAddress| {
+                assert(a == b, 'Incorrect address');
+            };
+
+            assert_eq_addresses(dispatcher.get_address(), contract_address);
+            assert_eq_addresses(get_contract_address(), test_address());
+
+            interact_with_state(
+                contract_address,
+                || {
+                    assert_eq_addresses(dispatcher.get_address(), contract_address);
+                    assert_eq_addresses(get_contract_address(), contract_address);
+
+                    // Make sure other contracts are not modified
+                    let other_dispatcher = IEmptyDispatcher { contract_address: other_empty_contract };
```

Comment:
I think it'd be good to also have a dispatcher created before `interact_with_state` and checking if it's address is not modified as well.

So we are verifying that `interact_with_state` doesn't affect previously created entities as well.